### PR TITLE
[#1870][#1898][#1922] improvement: dealing with unused fields

### PR DESCRIPTION
**Pull request by YxAc**
### What changes were proposed in this pull request?
- dealing with unused fields

### Why are the changes needed?
Fix: #1870 
Fix: #1898 
Fix: #1922 

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
NO

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorContext.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public CatalogConnectorContext(
     this.internalConnector = internalConnector;
     this.adapter = adapter;
 
-    this.connector = new GravitinoConnector(catalogName, this);
+    this.connector = new GravitinoConnector(this.catalogName, this);
```
Comment (YxAc) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorContext.java:45]: @justinmclean  I think `catalogName` is necessary for `GravitinoConnector`, which can not remove, so I change this by using `'this'`

**Comment by YxAc** [comment]
@justinmclean Pls take a look when you are free, thx.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorContext.java (hunk 1 above)
Comment (justinmclean) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorContext.java:45]: The is that while this.catalogName is set, it is never used and as far as I can see there no need for the field at all.

**Comment by justinmclean** [comment]
The issue also exists with these files:
entity in HadoopCatalogOperations.java
entity in JdbcCatalogOperations.java
entity in IcebergCatalogOperations.java
properties in HTTPClient.java
catalogName in CatalogConnectorContext.java

If you could fix those as well it would be a great help.

**Comment by YxAc** [comment]
> The issue also exists with these files: entity in HadoopCatalogOperations.java entity in JdbcCatalogOperations.java entity in IcebergCatalogOperations.java properties in HTTPClient.java catalogName in CatalogConnectorContext.java
> 
> If you could fix those as well it would be a great help.

OK, intuitively feel that entity in these classes is useful. I will take a closer look at these questions.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorContext.java (hunk 1 above)
Comment (justinmclean) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorContext.java:45]: I would remove the `catalogName` field as it is not used anywhere else that I can see

**Comment by jerryshao** [comment]
> > The issue also exists with these files: entity in HadoopCatalogOperations.java entity in JdbcCatalogOperations.java entity in IcebergCatalogOperations.java properties in HTTPClient.java catalogName in CatalogConnectorContext.java
> > If you could fix those as well it would be a great help.
> 
> OK, intuitively feel that entity in these classes is useful. I will take a closer look at these questions.

Yeah, I agree with @YxAc that `entity` is useful when we support managing all the entity metadata in Gravitino itself.

**Comment by YxAc** [comment]
> > > The issue also exists with these files: entity in HadoopCatalogOperations.java entity in JdbcCatalogOperations.java entity in IcebergCatalogOperations.java properties in HTTPClient.java catalogName in CatalogConnectorContext.java
> > > If you could fix those as well it would be a great help.
> > 
> > 
> > OK, intuitively feel that entity in these classes is useful. I will take a closer look at these questions.
> 
> Yeah, I agree with @YxAc that `entity` is useful when we support managing all the entity metadata in Gravitino itself.

hi @justinmclean , when I try to deal with unused fields problems, i think those entity are not a problem to fix as @jerryshao said. 
So May I only focus on this PR?

**Comment by jerryshao** [comment]
> > > > The issue also exists with these files: entity in HadoopCatalogOperations.java entity in JdbcCatalogOperations.java entity in IcebergCatalogOperations.java properties in HTTPClient.java catalogName in CatalogConnectorContext.java
> > > > If you could fix those as well it would be a great help.
> > > 
> > > 
> > > OK, intuitively feel that entity in these classes is useful. I will take a closer look at these questions.
> > 
> > 
> > Yeah, I agree with @YxAc that `entity` is useful when we support managing all the entity metadata in Gravitino itself.
> 
> hi @justinmclean , when I try to deal with unused fields problems, i think those entity are not a problem to fix as @jerryshao said. So May I only focus on this PR?

@YxAc yes, let's focus on this PR and make it done today. 😄

**Review by jerryshao — APPROVED** [review]
LGTM, thanks a lot @YxAc for your contribution.

**Pull request merged**